### PR TITLE
Update module documentation

### PR DESCRIPTION
**Pull request by smortex**
- Remove outdated recommandations from the README
- Update REFERENCE.md and remove it from REAME.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,13 +63,43 @@ This may be on the same host, or different hosts, but the name you put here
...
 
-##### Prefer hiera data
+```puppet
+class profile::bacula_client (
+  Sensitive $password,
+) {
+  class { 'bacula':
+    storage_name  => 'bacula.example.com',
+    director_name => 'bacula.example.com',
+    # Other common settings, see bellow
```
Comment (kenyon) [inline README.md:77]:
```suggestion
    # Other common settings, see below
```

File: README.md [hunk 2]
```diff
@@ -60,13 +63,43 @@ This may be on the same host, or different hosts, but the name you put here
...
+  }
+
+  class { 'bacula::client':
+    password => $password,
+    # File daemon specific settings, see bellow
```
Comment (kenyon) [inline README.md:82]:
```suggestion
    # File daemon specific settings, see below
```

File: README.md [hunk 3]
```diff
@@ -60,13 +63,43 @@ This may be on the same host, or different hosts, but the name you put here
...
+```
+
+```puppet
+class profile::bacula_server (
+) {
+  include profile::bacula_client
+
+  class { 'bacula::director':
+    password => 'director-password',
+    # Director specific settings, see bellow
```
Comment (kenyon) [inline README.md:94]:
```suggestion
    # Director specific settings, see below
```

File: README.md [hunk 4]
```diff
@@ -60,13 +63,43 @@ This may be on the same host, or different hosts, but the name you put here
...
+  }
 
-Users should prefer setting hiera data to set class parameter values where
-possible.  A couple calls in this module rely on hiera data present to avoid
-scoping issues associated with defined types and default values.
+  class { 'bacula::storage':
+    password => 'storage-password',
+    # File storage specific settings, see bellow
```
Comment (kenyon) [inline README.md:99]:
```suggestion
    # File storage specific settings, see below
```

File: README.md [hunk 5]
```diff
@@ -128,15 +165,17 @@ resource.
...
+```puppet
+class { 'bacula::client':
+  director => 'mydirector.example.com',
+}
 ```
 
 To direct all jobs to a specific pool like the one defined above set the
 following data.
 
 ```Puppet
```
Comment (kenyon) [inline README.md:177]: lowercase

**Review by bastelfreak — APPROVED** [review]

**Review by kenyon — APPROVED** [review]

**Pull request merged**